### ISSUE TITLE
Comments for Installing Git 2 on CentOS 7

### ISSUE THREAD

**Issue by jonjensen** [EndPointCorp/end-point-blog#1806]
Comments for https://www.endpointdev.com/blog/2021/12/installing-git-2-on-centos-7/
by Jon Jensen

To enter a comment:

1. Log in to GitHub
1. Leave a comment on this issue.

**Comment by grabear** [comment]
Hi @jonjensen.  Just curious how you were able to come up with these numbers:
>CentOS/RHEL 7 includes Git version 1.8.3, which was released in May 2013. There have been 2 major, 36 minor, and 216 patch releases of Git in the 8½ years between then and the current version 2.34.1!

**Comment by jonjensen** [comment]
@grabear I grabbed a list of Git release tags and excluded everything up to 1.8.3:

```bash
git tag | perl -ne 'next if !/^v/ or /-rc/; ++$started if /^v1\.8\.3$/; print if $started'
```

Then I counted the minor versions by grabbing those ending in `.0`:

```bash
git tag | perl -ne 'next if !/^v/ or /-rc/; ++$started if /^v1\.8\.3$/; print if $started' | grep -c '\.0$'
```

And the patch versions are all the releases:

```bash
git tag | perl -ne 'next if !/^v/ or /-rc/; ++$started if /^v1\.8\.3$/; print if $started' | wc -l
```

Perhaps I should have excluded the minor releases from the count of patch releases, which would be:

```bash
git tag | perl -ne 'next if !/^v/ or /-rc/; ++$started if /^v1\.8\.3$/; print if $started' | grep -cv '\.0$'
```

**Comment by grabear** [comment]
Thanks @jonjensen.  This was very helpful.  It will be nice to use this for other projects as well.

**Comment by corporate-gadfly** [comment]
@jonjensen  TIA for the repos. Will there be updated packages for git-2.35.0 and 2.35.1?

**Comment by jonjensen** [comment]
@corporate-gadfly Certainly not for both 2.35.0 and 2.35.1. We only build new packages of Git when there's a security or bugfix or feature change that we find worth the time. We're available for hire if you want something outside of our normal routine!

**Comment by corporate-gadfly** [comment]
https://github.blog/2022-04-12-git-security-vulnerability-announced/

**Comment by jonjensen** [comment]
@corporate-gadfly FYI we just released RHEL/CentOS 7 RPM packages for Git 2.36.0.

**Comment by robert914** [comment]
Any chance you could build and release new Git packages for version 2.37.0?
See https://github.blog/2022-06-27-highlights-from-git-2-37
Thanks!

**Comment by jonjensen** [comment]
@robert914 Yes. We now have Git 2.37.1 RPM packages in our CentOS 7 yum repository. Enjoy!

**Comment by ststing** [comment]
Would you add Git 2.39.1 package for CentOS? As mitigation these issues https://github.blog/2023-01-17-git-security-vulnerabilities-announced-2/

**Comment by jonjensen** [comment]
@ststing @corporate-gadfly We just published Git 2.39.1 packages for CentOS 7 & 8, FYI.

RHEL 9 family is coming soon.

**Comment by Sweatha-g** [comment]
Hi Would you add git 2.37.5 packages available for CENTOS7 - that's for https://confluence.atlassian.com/security/multiple-products-security-advisory-git-buffer-overflow-cve-2022-41903-cve-2022-23521-1189805967.html?utm_source=alert-email&utm_medium=email&utm_campaign=multi-product-advisory_february-2023_EML-15103&jobid=105935109&subid=1516357815

**Comment by jonjensen** [comment]
@Sweatha-g Please see past discussion above. No, we only freely build new packages of Git when there's a security or bugfix or feature change newer than the last version, that we find worth the time. We don't built patch release updates when a newer version already includes those fixes.

We are available for hire if you want something built outside of our normal routine!

**Comment by jonjensen** [comment]
Git 2.39.2 is now available in our RHEL/CentOS/Rocky/etc. 7 & 8 repositories.

**Comment by Sweatha-g** [comment]
2.37.5 is also an affected version

**Comment by jonjensen** [comment]
FYI we have Git 2.41.0 packages in our repositories at https://packages.endpointdev.com/

**Comment by dxb0523** [comment]
sudo yum -y install https://packages.endpointdev.com/rhel/7/os/x86_64/endpoint-repo.x86_64.rpm
Loaded plugins: fastestmirror, langpacks
Cannot open: https://packages.endpointdev.com/rhel/7/os/x86_64/endpoint-repo.x86_64.rpm. Skipping.
Error: Nothing to do

Any solution for the above error?

**Comment by jonjensen** [comment]
@dxb0523 Not sure why your system couldn't retrieve that file.

Does it still happen if you try again now? If so, what if you `curl` that file?

**Comment by joshwilliams** [comment]
@robert914 Are you looking for an updated Git package for RHEL / CentOS 7? The build system for that platform has been shut down now that it's EOL, I'm afraid. If you're looking for EL 8 or 9, 2.50.1 has been built and is available over on the packages site!

**Comment by robert914** [comment]
> [@robert914](https://github.com/robert914) Are you looking for an updated Git package for RHEL / CentOS 7? The build system for that platform has been shut down now that it's EOL, I'm afraid. If you're looking for EL 8 or 9, 2.50.1 has been built and is available over on the packages site!

Yes, I was looking for CentOS7 to support some old servers. Sorry to hear it was shutdown, but I understand. I'm working on rolling my own currently. Thanks.
